spec/greed_game_spec.rb

```diff
@@ -0,0 +1,56 @@
+require './greed_game'
+require './player'
+require './dice'
+
+RSpec.describe GreedGame do
+  let(:game) { GreedGame.new(2) }
+
+  describe '#initialize' do
+    it 'creates the correct number of players' do
+      expect(game.players.length).to eq(2)
+    end
+
+    it 'initializes players with the correct ids' do
+      expect(game.players.map(&:player_id)).to eq([1, 2])
+    end
+  end
+
+  describe '#calculate_score' do
+    it 'returns correct score for various dice rolls' do
```

Comment:
Pls add more edge cases like empty array, only single element in array, etc

Reply:
added